src/v/kafka/security/sasl_authentication.h

```diff
@@ -0,0 +1,75 @@
+// Copyright 2020 Vectorized, Inc.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.md
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0
+#pragma once
+#include "bytes/bytes.h"
+#include "vassert.h"
+
+#include <memory>
+
+namespace kafka {
+
+class sasl_error final : public std::exception {
```

Comment:
if the exceptions are small and specific (my hunch), we can do this with no memory alloations

simply create a few exception classes and return the literals

Reply:
there are only one or maybe two meaningful exception cases. everything else is free form protocol specific cases.

Reply:
was able to fully remove sasl_error